netsim-ffi/src/lib.rs

```diff
@@ -180,13 +183,15 @@ pub unsafe extern "C" fn netsim_socket_recv(
 
     *from = id;
 
-    // we need to take the minimum value between
-    // what the caller had allocated
+    if data.len() > output.len() {
+        return SimError::BufferTooSmall;
+    }
 
-    let copy_length = output.len().min(data.len()); // Determine the max length to copy
-    output[..copy_length].copy_from_slice(&data[..copy_length]);
+    let output_slice = &mut output[..data.len()];
+    // Copy data from `data` to `output_slice`
+    output_slice.copy_from_slice(&data);
 
-    *size = copy_length as u64;
+    *size = output_slice.len() as u64;
```

Comment:
I wonder if we couldn't make the case of copying all the data we can anyway and update the size accordingly anyway (with the size that would have been needed) and to return the error code bufferTooSmall. This way we give the a bit more information to the caller to debug their code if they didn't allocate enough buffer size.

Reply:
We should definitely return the size that's needed, that would help. 

I'm not sure what anyone will do with an incomplete message though. 
It might be misleading if they see *some* of their expected message in the buffer and they haven't checked the return code.